test/services/licence/caTransitions.test.ts

```diff
@@ -0,0 +1,486 @@
+      expect(allowed).toBe('caToDmRefusal')
+    })
+
+    test('should not allow CA to DM refusal if ineligible without', () => {
+      const status = {
+        stage: licenceStage.ELIGIBILITY,
+        postApproval: false,
+        tasks: {
+          exclusion: taskState.DONE,
+          crdTime: taskState.DONE,
+          suitability: taskState.DONE,
+          optOut: taskState.DONE,
+          bassReferral: taskState.DONE,
+          curfewAddress: taskState.DONE,
+          finalChecks: taskState.DONE,
+        },
+        decisions: {
+          eligible: null,
+          curfewAddressApproved: false,
+        },
+      }
+
+      const allowed = getAllowedTransition(status)
+      expect(allowed).toBe(null)
+    })
+
+    test('should not allow CA to DM refusal if already refused by ca', () => {
+      const status = {
+        stage: licenceStage.PROCESSING_CA,
+        postApproval: false,
+        tasks: {
+          exclusion: taskState.DONE,
+          crdTime: taskState.DONE,
+          suitability: taskState.DONE,
+          optOut: taskState.DONE,
+          bassReferral: taskState.DONE,
+          curfewAddress: taskState.DONE,
+          finalChecks: taskState.DONE,
+        },
+        decisions: {
+          curfewAddressApproved: false,
+          eligible: true,
+          finalChecksRefused: true,
+        },
+      }
+
+      const allowed = getAllowedTransition(status)
+      expect(allowed).toBe(null)
+    })
+
+    test('should allow CA to DM refusal if curfew address is rejected', () => {
+      const status = {
+        stage: licenceStage.ELIGIBILITY,
+        postApproval: true,
+        tasks: {
+          exclusion: taskState.DONE,
+          crdTime: taskState.DONE,
+          suitability: taskState.DONE,
+          optOut: taskState.DONE,
+          bassReferral: taskState.DONE,
+          curfewAddress: taskState.DONE,
+          finalChecks: taskState.DONE,
+        },
+        decisions: {
+          eligible: true,
+          curfewAddressRejected: true,
+        },
+      }
+
+      const allowed = getAllowedTransition(status)
+      expect(allowed).toBe('caToDmRefusal')
+    })
+
+    test('should allow CA to DM refusal if BASS area is rejected - POST_APPROVAL', () => {
+      const status = {
+        stage: licenceStage.MODIFIED,
+        postApproval: true,
+        tasks: {},
+        decisions: {
+          eligible: true,
+          bassReferralNeeded: true,
+          bassAreaNotSuitable: true,
+        },
+      }
+
+      const allowed = getAllowedTransition(status)
+      expect(allowed).toBe('caToDmRefusal')
+    })
+
+    test('should allow CA to DM refusal if BASS outcome is Unsuitable - POST_APPROVAL', () => {
+      const status = {
+        stage: licenceStage.MODIFIED,
+        postApproval: true,
+        tasks: {},
+        decisions: {
+          eligible: true,
+          bassReferralNeeded: true,
+          bassAreaSuitable: true,
+          bassAccepted: 'Unsuitable' as const,
```

Comment:
bit of a weird way to refer to members of union types.
Without this, we get a compilation error on the call to `getAllowedTransition` as it says that `string is not assignable to 'Unknown', 'Accepted', 'Unsuitable'`.

This is because:

```
let y = 'bob' // has a type of string
const x = 'bob' // has a type of 'bob'
```